server/src/main/java/org/elasticsearch/env/OverwriteNodeVersionCommand.java

```diff
@@ -0,0 +1,109 @@
+            "data from here into an older cluster.\n" +
+            "\n" +
+            "Do you want to proceed?\n";
+
+    static final String TOO_OLD_MESSAGE =
+        DELIMITER +
+            "\n" +
+            "This data path was last written by Elasticsearch version [V_OLD] which may be\n" +
+            "too old to be readable by Elasticsearch version [V_CUR].  This tool will bypass\n" +
+            "this compatibility check, allowing a version [V_CUR] node to start on this data\n" +
+            "path, but this version [V_CUR] node may not be able to read this data or may\n" +
+            "read it incorrectly leading to data loss.\n" +
+            "\n" +
+            "You should not use this tool. Instead, upgrade this data path from [V_OLD] to\n" +
+            "[V_CUR] using one or more intermediate versions of Elasticsearch.\n" +
+            "\n" +
+            "Do you want to proceed?\n";
+
+    static final String NO_METADATA_MESSAGE = "no node metadata found, so there is nothing to overwrite";
+    static final String SUCCESS_MESSAGE = "Successfully overwrote this node's metadata to bypass its version compatibility checks.";
+
+    public OverwriteNodeVersionCommand() {
+        super("Overwrite the version stored in this node's data path with [" + Version.CURRENT +
+            "] to bypass the version compatibility checks");
+    }
+
+    @Override
+    protected void processNodePaths(Terminal terminal, Path[] dataPaths, Environment env) throws IOException {
+        final Path[] nodePaths = Arrays.stream(toNodePaths(dataPaths)).map(p -> p.path).toArray(Path[]::new);
+        final NodeMetaData nodeMetaData = NodeMetaData.FORMAT.loadLatestState(logger, namedXContentRegistry, nodePaths);
+        if (nodeMetaData == null) {
+            throw new ElasticsearchException(NO_METADATA_MESSAGE);
+        }
+
+        try {
+            nodeMetaData.upgradeToCurrentVersion();
+            throw new ElasticsearchException("found [" + nodeMetaData + "] which is compatible with current version [" + Version.CURRENT
+                + "], so there is no need to overwrite it");
+        } catch (IllegalStateException e) {
+            // ok, means the version change is not supported
+        }
+
+        confirm(terminal, (nodeMetaData.nodeVersion().before(Version.CURRENT) ? TOO_OLD_MESSAGE : TOO_NEW_MESSAGE)
+            .replace("V_OLD", nodeMetaData.nodeVersion().toString())
+            .replace("V_NEW", nodeMetaData.nodeVersion().toString())
+            .replace("V_CUR", Version.CURRENT.toString()));
+
+        NodeMetaData.FORMAT.writeAndCleanup(new NodeMetaData(nodeMetaData.nodeId(), Version.CURRENT), nodePaths);
+
+        terminal.println(SUCCESS_MESSAGE);
+    }
+
+    //package-private for testing
+    OptionParser getParser() {
+        return parser;
+    }
+
+    //package-private for testing
+    void testExecute(Terminal terminal, OptionSet options, Environment env) throws Exception {
```

Comment:
My IDE claims this method is not used

Reply:
So does mine. Removed in 4961fd137f27601e3f89009280ee1095595f54aa.